processing/src/main/java/io/druid/segment/StringDimensionIndexer.java

```diff
@@ -509,7 +499,7 @@ public boolean matches()
       @Override
       public int getValueCardinality()
       {
-        return maxId;
+        return getCardinality();
```

Comment:
I think it'd be good to put something like this in `lookupName` too, as a sanity check:

```java
if (id >= maxId) {
  throw new ISE("id[%d] >= maxId[%d]", id, maxId);
}
```